management_api_app/db/repositories/resources.py

```diff
@@ -0,0 +1,26 @@
+from azure.cosmos import CosmosClient
+from pydantic import UUID4
+
+from core import config
+from db.errors import EntityDoesNotExist
+from db.repositories.base import BaseRepository
+from jsonschema import validate
+
+
+class ResourceRepository(BaseRepository):
+    def __init__(self, client: CosmosClient):
+        super().__init__(client, config.STATE_STORE_RESOURCES_CONTAINER)
+
+    @staticmethod
+    def _validate_resource_parameters(workspace_create, workspace_template):
+        validate(instance=workspace_create["properties"], schema=workspace_template)
+
+    def get_resource_dict_by_id(self, workspace_id: UUID4) -> dict:
```

Comment:
This doesn't look like a get_resource_dict_by_id but rather like a get_workspace_dict_by_id -- only time this makes sense is if it is a workspace (or should it say resource_id)

Reply:
Thanks for fixing the function name -- can we change workspace_id to be resource_id for consistency?